model-optimizer/extensions/back/SpecialNodesFinalization.py

```diff
@@ -130,6 +130,34 @@ def find_and_replace_pattern(self, graph: Graph):
             graph.remove_node(node.id)
 
 
+class RemoveConstToResult(BackReplacementPattern):
+    """
+    Transformation looks for a sub-graph "Const->Result" and removes Result node.
+    Currently IE is unable to handle such graph so this transformation removes to work around this case.
+    For instance, this case appears for Wide and Deep model.
+    """
+    enabled = True
+
+    @staticmethod
+    def pattern():
+        return dict(
+            nodes=[
+                ('const_node', {'type': 'Const', 'kind': 'op'}),
+                ('const_data', {'kind': 'data'}),
+                ('result_node', {'type': 'Result', 'kind': 'op'}),
+            ],
+            edges=[
+                ('const_node', 'const_data'),
+                ('const_data', 'result_node')
+            ]
+        )
+
+    @staticmethod
+    def replace_pattern(graph: Graph, match: dict):
+        result_node = match['result_node']
+        graph.remove_node(result_node.id)
```

Comment:
Here we remove only Result op node and leave graph in inconsistent form. We have everything we need to delete Const with its data node.

Reply:
It might be a case that const node is a part of another sub-graph. But if not, we can delete it.
@lazarevevgeny, please advice the optimal way to have consistent graph (force_cleanup, manual cleanup or stay as is).

Reply:
There is no guarantee that some other transformation will call "force_cleanup" after this one, so theoretically it may happen that the dangling Const will appear in the IR and cause an IE failure. Adding "force_cleanup" is quite expensive so I would recommend to add a check that there is just one consumer of "const_node" and if so then remove the "const_node" and "const_data". If there are more consumers of the "const_node" then most probably it is still connected with a Result op and nothing should be cleaned-up.

Reply:
Fixed.